src/main/java/net/consensys/orion/http/handler/set/SetPrivacyGroupHandler.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.http.handler.set;
+
+import static net.consensys.orion.http.server.HttpContentType.CBOR;
+
+import net.consensys.orion.enclave.PrivacyGroupPayload;
+import net.consensys.orion.enclave.QueryPrivacyGroupPayload;
+import net.consensys.orion.exception.OrionErrorCode;
+import net.consensys.orion.exception.OrionException;
+import net.consensys.orion.storage.Storage;
+import net.consensys.orion.utils.Serializer;
+
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Set a privacy group in the enclave, distinct from push as key is specified.
+ */
+public class SetPrivacyGroupHandler implements Handler<RoutingContext> {
+
+  private static final Logger log = LogManager.getLogger();
+
+  private final Storage<PrivacyGroupPayload> privacyGroupStorage;
+  private final Storage<QueryPrivacyGroupPayload> queryPrivacyGroupStorage;
+
+  public SetPrivacyGroupHandler(
+      final Storage<PrivacyGroupPayload> privacyGroupStorage,
+      final Storage<QueryPrivacyGroupPayload> queryPrivacyGroupStorage) {
+    this.privacyGroupStorage = privacyGroupStorage;
+    this.queryPrivacyGroupStorage = queryPrivacyGroupStorage;
+  }
+
+  @Override
+  @SuppressWarnings("rawtypes")
+  public void handle(final RoutingContext routingContext) {
+    final byte[] request = routingContext.getBody().getBytes();
+    final SetPrivacyGroupRequest addRequest = Serializer.deserialize(CBOR, SetPrivacyGroupRequest.class, request);
+    updatePrivacyGroupStorage(routingContext, addRequest);
```

Comment:
Question: Assumingly this will return a http response code of 500 if something throws internally (as per comment in the "Add" handler) - is that expected behaviour? Or will different failures require different codes?

Reply:
Yes, this is expected behavior at the moment. We'll look to improve this in future. It does need a bit of a cleanup.